boskos/client/client.go

```diff
@@ -71,12 +74,23 @@ type Client struct {
 //
 // Clients created with this function default to retrying failed connection
 // attempts three times with a ten second pause between each attempt.
-func NewClient(owner string, url string) *Client {
+func NewClient(owner string, url, username, password string) *Client {
+
+	getPassword := func() []byte { return []byte("") }
```

Comment:
Why do we need this?

Reply:
If no username is set we should just not call `getPassword()` or try to add auth to the request

Reply:
Done.
I was trying to avoid the nil check. :)